semver/range_test.ts

```diff
@@ -92,298 +97,382 @@ Deno.test("range", function () {
+      ["^1.2.3+build", "2.0.0"],
+      ["^1.2.3+build", "1.2.0"],
+      ["1.2.3-pre+asdf - 2.4.3-pre+asdf", "2.4.3"],
+      ["^1.2.3", "1.2.3-pre"],
+      ["^1.2", "1.2.0-pre"],
+      [">1.2", "1.3.0-beta"],
+
+      ["<=1.2.3", "1.2.4-beta"],
+      ["^1.2.3", "1.2.3-beta"],
+      ["<1.2.3", "1.2.3-beta"],
+      ["=1.2.3", "1.2.3-beta"],
 
-  versions.forEach(function (v) {
-    const range1 = new semver.Range(v[0]);
-    const range2 = new semver.Range(v[1]);
-    const expect = v[2];
-    const actual1 = range1.intersects(range2);
-    const actual2 = range2.intersects(range1);
-    const actual3 = semver.intersects(v[1], v[0]);
-    const actual4 = semver.intersects(v[0], v[1]);
-    const actual5 = semver.intersects(range1, range2);
-    const actual6 = semver.intersects(range2, range1);
-    assertEquals(actual1, expect);
-    assertEquals(actual2, expect);
-    assertEquals(actual3, expect);
-    assertEquals(actual4, expect);
-    assertEquals(actual5, expect);
-    assertEquals(actual6, expect);
-  });
+      ["=0.7.x", "0.7.0-asdf"],
+      [">=0.7.x", "0.7.0-asdf"],
+      ["1.0.0", "1.0.1"],
+      [">=1.0.0", "0.0.0"],
+      [">=1.0.0", "0.0.1"],
+      [">=1.0.0", "0.1.0"],
+      [">1.0.0", "0.0.1"],
+      [">1.0.0", "0.1.0"],
+      ["<=2.0.0", "3.0.0"],
+      ["<=2.0.0", "2.9999.9999"],
+      ["<=2.0.0", "2.2.9"],
+      ["<2.0.0", "2.9999.9999"],
+      ["<2.0.0", "2.2.9"],
+      [">=0.1.97", "0.1.93"],
+      ["0.1.20 || 1.2.4", "1.2.3"],
+      [">=0.2.3 || <0.0.1", "0.0.3"],
+      [">=0.2.3 || <0.0.1", "0.2.2"],
+      ["2.x.x", "3.1.3"],
+      ["1.2.x", "1.3.3"],
+      ["1.2.x || 2.x", "3.1.3"],
+      ["1.2.x || 2.x", "1.1.3"],
+      ["2.*.*", "1.1.3"],
+      ["2.*.*", "3.1.3"],
+      ["1.2.*", "1.3.3"],
+      ["1.2.* || 2.*", "3.1.3"],
+      ["1.2.* || 2.*", "1.1.3"],
+      ["2", "1.1.2"],
+      ["2.3", "2.4.1"],
+      ["~0.0.1", "0.1.0-alpha"],
+      ["~0.0.1", "0.1.0"],
+      ["~2.4", "2.5.0"], // >=2.4.0 <2.5.0
+      ["~2.4", "2.3.9"],
+      ["~>3.2.1", "3.3.2"], // >=3.2.1 <3.3.0
+      ["~>3.2.1", "3.2.0"], // >=3.2.1 <3.3.0
+      ["~1", "0.2.3"], // >=1.0.0 <2.0.0
+      ["~>1", "2.2.3"],
+      ["~1.0", "1.1.0"], // >=1.0.0 <1.1.0
+      ["<1", "1.0.0"],
+      [">=1.2", "1.1.1"],
+      ["~v0.5.4-beta", "0.5.4-alpha"],
+      ["=0.7.x", "0.8.2"],
+      [">=0.7.x", "0.6.2"],
+      ["<0.7.x", "0.7.2"],
+      [">1.2", "1.2.8"],
+      ["^0.0.1", "0.0.2-alpha"],
+      ["^0.0.1", "0.0.2"],
+      ["^1.2.3", "2.0.0-alpha"],
+      ["^1.2.3", "1.2.2"],
+      ["^1.2", "1.1.9"],
+
+      // invalid ranges never satisfied!
+      ["blerg", "1.2.3"],
+      ["^1.2.3", "2.0.0-pre"],
+    ];
+
+    for (const [r, v] of versions) {
+      await t.step(`${r} ∌ ${v}`, () => {
+        const range = parseRange(r);
+        const s = parse(v);
+        const found = rangeTest(s, range);
+        assert(!found);
+      });
+    }
+  },
 });
+
+// Deno.test("unlockedPrereleaseRange", function () {
```

Comment:
Should these commented out test cases be removed?

Reply:
No, it looks like I missed them and I need to fix em up and get them running again.